### ISSUE TITLE
[Settings]: Radio buttons are not associated with its main label throughout the settings page.

### ISSUE THREAD

**Issue by v-rpundir** [microsoft/terminal#11248]
### Windows Terminal version (or Windows build number)

1.10.2383.0

### Other Software

Test Environment:
OS: Windows 11 Version 21H2 (OS Build 22000.168)
Screen Reader: Narrator
Tool: Accessibility Insight for Windows

### Steps to reproduce

Repro Steps:

1. Open Windows Terminal.
2. Open Settings page using "Ctrl + ',' ".
3. Open Narrator using 'Win + Ctrl+ Enter' key.
3. Navigate through all radio groups through Settings page and observe the issue.

**User Experience:**
Screen Reader user are impacted here as they will not be able to know the exact purpose of radio buttons as main label is not announced to them.  

**Guideline Reference:**
https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships

**Attachments:**
[Radio buttons are not associated with its main label..zip](https://github.com/microsoft/terminal/files/7176480/Radio.buttons.are.not.associated.with.its.main.label.zip)

### Expected Behavior

Radio buttons should be associated with its main label throughout the Settings page. i.e. For 'Default' radio button with group label 'Launch mode' Screen Reader should announce 'Launch mode group Default radio button selected 1 of 5' 

### Actual Behavior

Radio buttons is not associated with its main label throughout the Settings page. i.e. For 'Default' radio button with group label 'Launch mode' Screen Reader is announcing 'Default radio button selected 1 of 5' 

Observation: Same issue repro with NVDA Screen Reader.

**Comment by zadjii-msft** [comment]
This looks like #9567 which is actually tracked by the upstream https://github.com/microsoft/microsoft-ui-xaml/issues/3183. @v-rpundir can you confirm?

**Comment by v-rpundir** [comment]
@zadjii-msft  Yeah it is similar to #9567 but the Upstream bug is closed as fixed now.

**Comment by zadjii-msft** [comment]
Looks like the upstream bug was fixed in https://github.com/microsoft/microsoft-ui-xaml/pull/5094 which was in WinUI 2.6.1, so hopefully the 1.12 builds of the Terminal (which moved to WinUI 2.7) should have the fix. 

Looks like it's still not grouping them though:
![image](https://user-images.githubusercontent.com/18356694/136196083-515ffb9b-e051-4a36-8d3f-c64faf372a9f.png)

Maybe there's something else we need to do to manually combine these now.

**Comment by DHowett** [comment]
> Looks like it's still not grouping them though:


I suspect we need to do something special with SettingContainer, _assuming that this is using it_

**Comment by zadjii-msft** [comment]
@v-rpundir I believe this was fixed in `Preview v1.12.2922.0`/`Stable v1.11.2921.0`, could you confirm?

**Comment by v-rpundir** [comment]
Closing as the bug as fixed in Windows Terminal Preview Version: 1.12.3472.0

**Comment by Priyanka-Chauhan123** [comment]
Closing the bug as UI has been changed.
